src/kernels/jupyter/jupyterUtils.ts

```diff
@@ -56,7 +56,30 @@ export async function handleCertsError(
     );
     if (value === enableOption) {
         sendTelemetryEvent(Telemetry.SelfCertsMessageEnabled);
-        void config.updateSetting('allowUnauthorizedRemoteConnection', true, undefined, ConfigurationTarget.Workspace);
+        await config.updateSetting('allowUnauthorizedRemoteConnection', true, undefined, ConfigurationTarget.Workspace);
+        return true;
+    } else if (value === closeOption) {
+        sendTelemetryEvent(Telemetry.SelfCertsMessageClose);
+    }
+    return false;
+}
+
+export async function handleExpiredCertsError(
```

Comment:
The certs we have in the repo are expired. Figured expired certificate is still a valid test case and a user might do this too.